enhancements/machine-api/machine-api-usage-telemetry.md

```diff
@@ -0,0 +1,340 @@
+unable to perform scaling operations. To maximize the value of this data, we
+will want to report categories based on the different types of failures.
+Events related to min/max threshold on MachineSets probably don’t need
+alerting, but the inability to scale for other reasons (eg, hitting max CPU threshold or
+memory, but haven’t hit max node count) are useful to collect. These events can
+show opportunities to inform users about resource limitations in their clusters.
+
+**Machines with Old Delete Timestamp**
+
+Epics: Proactive Support
+
+This metric can show us stale resources that exist within clusters. By collecting
+data for machines that are stuck in a deletion phase for prolonged periods of time
+we can identify opportunities to inform users and create alerts around these
+stale resources. This alerting might need to be tunable (if possible), but a
+sane default might be on the order of 6 hours. When combined with a count of
+machines that contains old deletion timestamps this can become a powerful alarm
+for users. A starting threshold of 25% of machines with deletion timestamps older
+than 6 hours is the suggested line for an alarm.
+
+**MachineSets Usage**
+
+Epics: Engineering Insights
+
+This metric will give us information about how the clusters are being used,
+how users are interacting with the various APIs, and where user experience
+improvements can be targeted. By collecting data on the number and types of
+Machines being utilized, and how many MachineSets are created and destroyed by
+users we will learn how users operate the Machine API. This data can provide
+a multitude of insights into API and platform usage patterns.
+
+For example, seeing that users are creating many different Machines of varying
+flavors through multiple MachineSets will give us insight into how users would
+prefer to consume cloud resources. Conversely, seeing a high number of Machine
+creations without corresponding MachineSets might indicate that the API is not
+clearly understood, or that Machine resource sizing is less important to users.
+
+**Under-utilized Cluster Autoscaler**
+
+Epics: Engineering Insights, Proactive Support
+
+This metric will help us identify when users have enabled the Cluster Autoscaler
+but are not using it. By collecting data on the number of MachineAutoscalers
+that have been deployed in a cluster correlated with the deployment of the
+Cluster Autoscaler we can identify under-utilization cases. This data
+shows us opportunties to inform users and increase education around usage of
+the autoscaler.
+
+#### Alerts, Metrics, and Telemetry
+
+To fully accomplish the goals proposed in this enhancement several metrics
+will need to be exposed through telemetry. These metrics are:
+
+* Machine resource count
+* MachineSet resource count
+* MachineAutoscaler resource count
+* MachineHealthCheck resource count
+* MachineHealthCheck remediation success count
```

Comment:
these should not be sent separately, but instead via an alert when the failure rate is too high to be acceptable

Reply:
that makes sense to me, and i was planning that we would use the alerts to augment this data.

would it be acceptable to return total remediations? (i'm just thinking about how do we calculate the remediation effectiveness without know both total attempts and failures)

Reply:
yes I think that would be ok, try to aim for a maximum of 10 series aside from alerts

Reply:
i am working on adding more information about the specific metrics and series we would like to generate from this enhancement. hopefully that will help add more context to the specifics (cc @lilic )